script.py

```diff
@@ -17,17 +26,19 @@ def run() -> tuple[bool, str]:
             - The first element says whether the test run was successful or not.
             - The second element is a message shown in the Github output.
     """
-    if not (test_id and cj_token):
+    if not (test_id and service_account_key):
         return False, "Failed: All required fields were not provided."
 
-    headers = {
-        "Authorization": f"Bearer {cj_token}",
-        "Content-Type": "application/json"
-    }
+    # Login the serive account
+    response = requests.post(f"{BACKEND_URL}/auth/login_service_account",
+                             headers=_get_headers(service_account_key))
+    if response.status_code != 200:
+        return False, f"Failed to login service account: {response.json()}"
 
-    test_run_ep_stub = "https://cj-backend.foreai.co/test-run"
+    jwt_token = response.text.replace('"', "")
```

Comment:
Would just getting `response.json()` work here?

Reply:
It does, thanks!